Auto-unboxing issue in CheckstyleAuditListener

Sonar version: 7.6
Checkstyle sonar plugin verson: 4.18

Steps to reproduce or description of problem:

Run any check which reports line 0 for an issue. This used to be the required behavior for checks that flag the entire file, and have no relation to the AST. For example FileSetChecks.  
I know there was a breaking change in checkstyle/checkstyle#6000, changing the "magic" line number to 1. This kind of neutralizes this problem.  
BUT: I noticed the problem running custom checks at work. I would expect some custom checks to be around that haven't noticed checkstyle/checkstyle#6000.

The error caused is this:
```
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing C:\path\to\File.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:304)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:217)
        at org.sonar.plugins.checkstyle.CheckstyleExecutor.executeWithClassLoader(CheckstyleExecutor.java:111)
        ... 71 more
Caused by: java.lang.NullPointerException
        at org.sonar.plugins.checkstyle.CheckstyleAuditListener.addError(CheckstyleAuditListener.java:106)
        at com.puppycrawl.tools.checkstyle.Checker.fireErrors(Checker.java:401)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:292)
        ... 73 more
```

I have already fixed this, it's a small thing. PR upcoming.

--- a/.gitignore
+++ b/.gitignore
@@ -20,6 +20,7 @@ Icon?
 # Files that might appear on external disk
 .Spotlight-V100
 .Trashes
+_support/
 
 # ---- Windows
 # Windows image file caches

--- a/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleAuditListener.java
+++ b/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleAuditListener.java
@@ -100,15 +100,10 @@ public class CheckstyleAuditListener implements AuditListener {
             final NewIssue issue = context.newIssue();
             final ActiveRule rule = ruleFinder.find(
                     RuleKey.of(CheckstyleConstants.REPOSITORY_KEY, ruleKey));
-            final Integer lineId = getLineId(event);
-            int lineNo = 1;
-            if (lineId != null) {
-                lineNo = lineId;
-            }
             if (Objects.nonNull(issue) && Objects.nonNull(rule)) {
                 final NewIssueLocation location = issue.newLocation()
                         .on(currentResource)
-                        .at(currentResource.selectLine(lineNo))
+                        .at(currentResource.selectLine(getLineId(event)))
                         .message(message);
                 issue.forRule(rule.ruleKey())
                         .at(location)
@@ -162,14 +157,12 @@ public class CheckstyleAuditListener implements AuditListener {
     }
 
     @VisibleForTesting
-    static Integer getLineId(AuditEvent event) {
-        Integer result = null;
+    static int getLineId(AuditEvent event) {
+        int result = 1;
         try {
-            final int line = event.getLine();
-            // checkstyle returns 0 if there is no relation to a file content,
-            // but we use null
-            if (line != 0) {
-                result = line;
+            final int eventLine = event.getLine();
+            if (eventLine > 0) {
+                result = eventLine;
             }
         }
         catch (Exception ex) {

--- a/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleAuditListener.java
+++ b/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleAuditListener.java
@@ -100,10 +100,15 @@ public class CheckstyleAuditListener implements AuditListener {
             final NewIssue issue = context.newIssue();
             final ActiveRule rule = ruleFinder.find(
                     RuleKey.of(CheckstyleConstants.REPOSITORY_KEY, ruleKey));
+            final Integer lineId = getLineId(event);
+            int lineNo = 1;
+            if (lineId != null) {
+                lineNo = lineId;
+            }
             if (Objects.nonNull(issue) && Objects.nonNull(rule)) {
                 final NewIssueLocation location = issue.newLocation()
                         .on(currentResource)
-                        .at(currentResource.selectLine(getLineId(event)))
+                        .at(currentResource.selectLine(lineNo))
                         .message(message);
                 issue.forRule(rule.ruleKey())
                         .at(location)

--- a/checkstyle-sonar-plugin/src/test/java/org/sonar/plugins/checkstyle/CheckstyleAuditListenerTest.java
+++ b/checkstyle-sonar-plugin/src/test/java/org/sonar/plugins/checkstyle/CheckstyleAuditListenerTest.java
@@ -86,7 +86,7 @@ public class CheckstyleAuditListenerTest {
 
         eventTest = new AuditEvent(this, "", new LocalizedMessage(0, "", "", null, "",
                 CheckstyleAuditListenerTest.class, "msg"));
-        assertThat(CheckstyleAuditListener.getLineId(eventTest)).isNull();
+        assertThat(CheckstyleAuditListener.getLineId(eventTest)).isEqualTo(1);
         assertThat(CheckstyleAuditListener.getMessage(eventTest)).isEqualTo("msg");
         assertThat(CheckstyleAuditListener.getRuleKey(eventTest)).isEqualTo(
                 CheckstyleAuditListenerTest.class.getName());
@@ -99,19 +99,33 @@ public class CheckstyleAuditListenerTest {
                 CheckstyleAuditListenerTest.class.getName());
 
         eventTest = new AuditEvent(this);
-        assertThat(CheckstyleAuditListener.getLineId(eventTest)).isNull();
+        assertThat(CheckstyleAuditListener.getLineId(eventTest)).isEqualTo(1);
         assertThat(CheckstyleAuditListener.getMessage(eventTest)).isNull();
         assertThat(CheckstyleAuditListener.getRuleKey(eventTest)).isNull();
 
         eventTest = new AuditEvent(this, "", new LocalizedMessage(0, "", "", null, "module",
                 CheckstyleAuditListenerTest.class, "msg"));
-        assertThat(CheckstyleAuditListener.getLineId(eventTest)).isNull();
+        assertThat(CheckstyleAuditListener.getLineId(eventTest)).isEqualTo(1);
         assertThat(CheckstyleAuditListener.getMessage(eventTest)).isEqualTo("msg");
         assertThat(CheckstyleAuditListener.getRuleKey(eventTest)).isEqualTo("module");
     }
 
     @Test
     public void addErrorTest() {
+        addErrorTestForLine(42);
+    }
+
+    @Test
+    public void addErrorLine0Test() {
+        addErrorTestForLine(0);
+    }
+
+    @Test
+    public void addErrorLine1Test() {
+        addErrorTestForLine(1);
+    }
+
+    private void addErrorTestForLine(final int pLineNo) {
         final ActiveRule rule = setupRule("repo", "key");
 
         final NewIssue newIssue = mock(NewIssue.class);
@@ -125,9 +139,12 @@ public class CheckstyleAuditListenerTest {
         when(newLocation.message(anyString())).thenReturn(newLocation);
 
         when(inputFile.selectLine(anyInt())).thenReturn(new DefaultTextRange(
-                new DefaultTextPointer(1, 1), new DefaultTextPointer(1, 2)));
+            new DefaultTextPointer(1, 1), new DefaultTextPointer(1, 2)));
 
-        addErrorToListener(event);
+        final AuditEvent eventAdded = new AuditEvent(this, file.getAbsolutePath(),
+            new LocalizedMessage(pLineNo, "", "", null, "", CheckstyleAuditListenerTest.class,
+                "msg"));
+        addErrorToListener(eventAdded);
 
         verify(newIssue, times(1)).save();
         verify(newIssue, times(1)).forRule(rule.ruleKey());

